core/src/main/java/hudson/security/AccessDeniedException2.java

```diff
@@ -1,50 +1,75 @@
 package hudson.security;
 
-import org.acegisecurity.AccessDeniedException;
-import org.acegisecurity.Authentication;
-import org.acegisecurity.GrantedAuthority;
-
 import javax.servlet.http.HttpServletResponse;
 import java.io.PrintWriter;
 import jenkins.util.SystemProperties;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
 
 /**
- * {@link AccessDeniedException} with more information.
+ * {@link org.acegisecurity.AccessDeniedException} with more information.
  * @author Kohsuke Kawaguchi
  */
-public class AccessDeniedException2 extends AccessDeniedException {
+public class AccessDeniedException2 extends org.acegisecurity.AccessDeniedException {
```

Comment:
Deriving from the Acegi Security version in hopes of offering some compatibility, but untested.